src/config.rs

```diff
@@ -532,6 +543,369 @@ impl Config {
+                    std::process::exit(ERROR_STATUS_FUSE);
+                }
+                Input::File(input_source)
+            }
+            None => {
+                error!("The directory to pack must be specified.");
+                std::process::exit(ERROR_STATUS_CLI);
+            }
+        };
+
+        // set the mount from the input directory
+        config.mount = match &config.input {
+            Input::File(file) => Some(file.clone().canonicalize().unwrap()),
+            _ => {
+                error!("Input must be a file or directory.");
+                std::process::exit(ERROR_STATUS_CLI);
+            }
+        };
+
+        // configure output
+        config.output = if let Some(output) = args.value_of("OUTPUT") {
+            Output::File(PathBuf::from(output))
+        } else if args.is_present("NOOUTPUT") || args.is_present("QUIET") {
+            Output::Quiet
+        } else {
+            Output::Stdout
+        };
+
+        // try to autodetect the output format.
+        //
+        // first see if it's specified and parses okay.
+        //
+        // then see if we can pull it out of the extension (if specified)
+        //
+        // then give up and use the input format
+        config.output_format = match args
+            .value_of("TARGET_FORMAT")
+            .ok_or(format::ParseFormatError::NoFormatProvided)
+            .and_then(|s| s.parse::<Format>())
+        {
+            Ok(target_format) => target_format,
+            Err(e) => {
+                match e {
+                    format::ParseFormatError::NoSuchFormat(s) => {
+                        warn!(
+                            "Unrecognized format '{}', inferring from input and output.",
+                            s
+                        )
+                    }
+                    format::ParseFormatError::NoFormatProvided => {
+                        debug!("Inferring output format from input.")
+                    }
+                };
+                match args
+                    .value_of("OUTPUT")
+                    .and_then(|s| Path::new(s).extension())
+                    .and_then(|s| s.to_str())
+                {
+                    Some(s) => match s.parse::<Format>() {
+                        Ok(format) => format,
+                        Err(_) => {
+                            warn!(
+                                "Unrecognized format {}, defaulting to input format '{}'.",
+                                s, config.input_format
+                            );
+                            config.input_format
+                        }
+                    },
+                    None => config.input_format,
+                }
+            }
+        };
+
+        if config.pretty && !config.output_format.can_be_pretty() {
+            warn!(
+                "There is no pretty printing routine for {}.",
+                config.output_format
+            )
+        }
+
+        config
+    }
+
```

Comment:
How much of these can be made in common? (Not asking you to do it, but probably a good cleanup PR for you or me to hit next.)